liquid_benchmarks/driver.rb

```diff
@@ -0,0 +1,142 @@
+#
+# Liquid Benchmark driver
+#
+require 'digest'
+require 'net/http'
+require 'json'
+require 'pathname'
+require 'optparse'
+
+RAW_URL = 'https://raw.githubusercontent.com/ruby-bench/ruby-bench-suite/master/liquid/benchmarks/'
+
+class BenchmarkDriver
+  def self.benchmark(options)
+    self.new(options).run
+  end
+
+  def initialize(options)
+    @repeat_count = options[:repeat_count]
+    @pattern = options[:pattern]
+    @local = options[:local]
+  end
+
+  def run
+    files.each do |path|
+      next if !@pattern.empty? && /#{@pattern.join('|')}/ !~ File.basename(path)
+      run_single(path)
+    end
+  end
+
+  private
+
+  def files
+    Pathname.glob("#{File.expand_path(File.dirname(__FILE__))}/bm_*")
+  end
+
+  def run_single(path)
+    script = "ruby #{path}"
+
+    output = measure(script)
+    return unless output
+
+    request = Net::HTTP::Post.new('/benchmark_runs')
+    request.basic_auth(ENV["API_NAME"], ENV["API_PASSWORD"])
+
+    initiator_hash = {}
+    if(ENV['RUBY_COMMIT_HASH'])
+      initiator_hash['commit_hash'] = ENV['RUBY_COMMIT_HASH']
+    elsif(ENV['RUBY_VERSION'])
+      initiator_hash['version'] = ENV['RUBY_VERSION']
+    end
+
+    submit = {
+      'benchmark_type[category]' => "Liquid #{output['label']}",
+      'benchmark_type[script_url]' => "#{RAW_URL}#{path.basename}",
+      'benchmark_type[digest]' => generate_digest(path),
+      'benchmark_run[environment]' => "#{`ruby -v; gem -v`.strip}",
+      'repo' => 'ruby',
+      'organization' => 'ruby'
+    }.merge(initiator_hash)
+
+    request.set_form_data(submit.merge(
+      {
+        "benchmark_run[result][iterations_per_second]" => output["iterations_per_second"].round(3),
+        'benchmark_result_type[name]' => 'Number of iterations per second',
+        'benchmark_result_type[unit]' => 'Iterations per second'
+      }
+    ))
+
+    endpoint.request(request) unless @local
```

Comment:
Looks like you're submitting the same request object twice. Is that intentional?